sdf/1.7/frame.sdf

```diff
@@ -1,16 +1,32 @@
 <!-- Frame -->
 <element name="frame" required="*">
-  <description>A frame of reference to which a pose is relative.</description>
+  <description>A frame of reference in which poses may be expressed.</description>
 
   <attribute name="name" type="string" default="" required="1">
-    <description>Name of the frame. This name must not match another frame defined inside the parent that this frame is attached to.</description>
+    <description>
+      Name of the frame. It must be unique whithin its scope (model/world),
+      i.e., it must not match the name of another frame, link, joint, or model
+      within the same scope.
+    </description>
   </attribute>
 
   <attribute name="attached_to" type="string" default="" required="*">
     <description>
-      Name of the link or frame to which this frame is attached.
-      If a frame is specified, recursively following the attached_to attributes
-      of the specified frames must lead to the name of a link, a model, or the world frame.
+      If specified, this frame is attached to the specified frame. The specified
+      frame must be within the same scope and may be defined implicitly, i.e.,
+      the name of any //frame, //model, //joint, or //link within the same scope
+      may be used.
+
+      If missing, this frame is attached to the containing scope's frame. Within
+      a //world scope this is the implicit world frame, and within a //model
+      scope this is the implicit frame defined by the model's canonical link.
```

Comment:
The model's implicit frame (aka `__model__`) is different from the implicit frame of the canonical link. For example,
```xml
<model name='M'>
  <link name='L'>
    <pose>1 0 0   0 0 0</pose>
   </link>
</model>
```
The canonical link `L` is offset by pose `[1, 0, 0, 0, 0, 0]` from the model frame. So I think it would be better to say "...within a //model scope this is the implicit model frame".

Reply:
Hm. That part is not quite clear to me yet. If I don't specify `@attached_to`  for a frame it will be attached to `__model__`  instead of being attached to the model's canonical link?

I totally get that the frame's pose will `@relative_to`  the `__model__` frame, but I was under the impression that `__model__` is only used for initial placement and not for dynamics. Isn't the canonical link the element that gets (dynamically) connected to the parent model/world? Then again, I might misunderstand the concept/purpose of the canonical link element.

@azeey I will update it as per your suggestion. At the same time, it would help me if you could comment on what I just said so that I can figure out what I am confusing.

Reply:
> Hm. That part is not quite clear to me yet. If I don't specify @attached_to for a frame it will be attached to __model__ instead of being attached to the model's canonical link?

Yes, but note that the `__model__` frame in turn is attached to the canonical link.

> I totally get that the frame's pose will @relative_to the __model__ frame, but I was under the impression that __model__ is only used for initial placement and not for dynamics.

No, the `__model__` frame is the implicit frame of the model, so it's not just for initial placement. The canonical link is the rigid body that the `__model__` frame is attached to since all frames, implicit or explicit, must terminate on a rigid body.

There's a debugging tool that visualizes the frame graph. I think that might help to clarify.

```
xdot <(ign sdf -g frame test/sdf/model_frame_attached_to.sdf)
```